frontend/simple/js/event-log.js

```diff
@@ -0,0 +1,107 @@
+import localforage from 'localforage'
+import {toHash, makeEntry, makeEvent, makeLog} from '../../../shared/functions'
+import request from 'superagent'
+import {EVENTS} from '../../../shared/events'
+import {Group} from '../../../shared/types'
+const {CREATION} = EVENTS
+var store
+// =======================
+// Helpers
+// =======================
+let _db = new Map()
+function getDB (hash: string) {
+  if (_db.has(hash)) {
+    return _db.get(hash)
+  }
+  let db = localforage.createInstance({
+    name: 'Group Income',
+    storeName: hash
+  })
+  _db.set(hash, db)
+  return db
+}
+// =======================
+// Exports
+// =======================
+// getItemFromLog retrieves from the log
+// opts object allows for parent hash to be returned instead of data
+export default {
+  async getItemFromLog (hash: string) {
+    if (!store.state.loggedInUser) {
+      return
+    }
+    let groupId = store.state.session.currentGroup.groupId
+    let db = getDB(groupId)
+    let entry = await db.getItem(hash)
+    return entry
+  },
+  // addItemToLog appends to the log
+  // returns the current state of the log
+  async addItemToLog (value: any) {
+    if (!store.state.loggedInUser) {
+      return
+    }
+    let groupId = store.state.session.currentGroup.groupId
+    let db = getDB(groupId)
+    let currentLogPosition = await db.getItem('currentLogPosition')
+    let entry = makeEntry(value, currentLogPosition)
+    let hash = toHash(entry)
+    await request.post(`${process.env.API_URL}/event/${groupId}`)
+        .send({ hash, entry })
+    await db.setItem(hash, entry)
+    await db.setItem('currentLogPosition', hash)
+    store.commit('updateCurrentLogPosition', hash)
```

Comment:
I'm not sure that makes sense to dispatch and additional actions since every one of these event log function is initially called by an action. The return to the action will handle the session debounce.  After determining that it would take multiple variables to returned to some action for them to commit all the mutations I thought it best to just do it from  event-log because the actions are what call them and they appear as the same state changes no matter where the commit comes from.